btc/src/main/kotlin/provider/btc/generation/BtcPublicKeyProvider.kt

```diff
@@ -72,48 +73,56 @@ class BtcPublicKeyProvider(
      * Creates multisignature address if enough public keys are provided
      * @param notaryKeys - list of all notaries public keys
      * @param addressType - type of address to create
+     * @param generationTime - time of address generation. Used in Iroha multisig
      * @return Result of operation
      */
     fun checkAndCreateMultiSigAddress(
         notaryKeys: Collection<String>,
-        addressType: BtcAddressType
+        addressType: BtcAddressType,
+        generationTime: Long
     ): Result<Unit, Exception> {
         return Result.of {
             val peers = notaryPeerListProvider.getPeerList().size
             if (peers == 0) {
                 throw IllegalStateException("No peers to create btc multisignature address")
-            } else if (notaryKeys.size == peers && hasMyKey(notaryKeys)) {
-                val threshold = getSignThreshold(peers)
-                val msAddress = createMsAddress(notaryKeys, threshold)
-                if (!walletFile.wallet.addWatchedAddress(msAddress)) {
-                    throw IllegalStateException("BTC address $msAddress was not added to wallet")
-                }
-                logger.info { "Address $msAddress was added to wallet. Current wallet state:\n${walletFile.wallet}" }
+            } else if (notaryKeys.size != peers) {
+                logger.info { "Not enough keys are collected to generate a multisig address(${notaryKeys.size} out of ${peers})" }
+                return@of
```

Comment:
Don't we need to throw something instead of just `return@of` ?

Reply:
No. It simply means that we in the middle of the process.